packages/pg/lib/stream.js

```diff
@@ -18,11 +18,32 @@ module.exports.getStream = function getStream(ssl) {
  * @returns {Duplex}
  */
 module.exports.getSecureStream = function getSecureStream(options) {
-  var tls = require('tls')
-  if (tls.connect) {
-    return tls.connect(options)
-  } else {
+  if (isCloudflareRuntime) {
     options.socket.startTls(options)
     return options.socket
+  } else {
+    var tls = require('tls')
+    return tls.connect(options)
+  }
+}
+
+/**
+ * Are we running in a Cloudflare Worker?
+ *
+ * @returns true if the code is currently running inside a Cloudflare Worker.
+ */
+function isCloudflareRuntime() {
```

Comment:
can you memoize this function please? otherwise it would run many times unnecessarily.``
```suggestion
const isCloudflareRuntime = (function isCloudflareRuntimeInit() {
```